### PR TITLE
Click on CSS button to refresh the page instead manual page refresh

### PR DESCRIPTION
**Pull request by meaksh**
This PR fixes an issue caused due a waiting step which refreshes the page in a wrong way (without keeping the package list filter). 🍰  

These changes will go to `manager31` and `master` branches.

### CODE REVIEW

File: features/step_definitions/salt_steps.rb [hunk 1]
```diff
@@ -459,10 +459,29 @@
       end
     end
   rescue Timeout::Error
-    raise "'#{arg1}' cannot be found after several tries"
+    raise "'#{arg1}' is still found after several tries"
   end
   raise unless not_found
 end
+
+Then(/^I click on the css "(.*)" until page does contain "([^"]*)" text$/) do |css, arg1|
```
Comment (MalloZup) [inline features/step_definitions/salt_steps.rb]: can we put "text" instead of arg1?

Comment (meaksh) [inline features/step_definitions/salt_steps.rb]: Yes, we can! 👍 😄

File: features/step_definitions/salt_steps.rb [hunk 2]
```diff
@@ -459,10 +459,29 @@
...
+  found = false
+  begin
+    Timeout.timeout(30) do
```
Comment (MalloZup) [inline features/step_definitions/salt_steps.rb]: we could use DEFAULT_TIMEOUT here

Comment (MalloZup) [inline features/step_definitions/salt_steps.rb]: or we could make MIN_DEFAULT_TIMEOUT label instead of magic numbers..

**Comment by MalloZup** [comment]
robo

File: features/step_definitions/salt_steps.rb [hunk 3]
```diff
@@ -446,23 +446,42 @@
...
+        unless page.has_content?(text)
           not_found = true
           break
         end
         find(css).click
       end
     end
   rescue Timeout::Error
-    raise "'#{arg1}' cannot be found after several tries"
+    raise "'#{text}' still found after several tries"
```
Comment (ncounter) [inline features/step_definitions/salt_steps.rb]: I think the meaning should be "not yet found`, am I getting it wrong?

Comment (meaksh) [inline features/step_definitions/salt_steps.rb]: Not in this case. This particular step is actually failing if `#{text}` is still found after the timeout is reached.

The case you're mentioning is just covered by the next step implemented on this file: https://github.com/SUSE/spacewalk-testsuite-base/pull/387/files#diff-daf5562d26751e108a56ac4ad5257bc9R480

Comment (ncounter) [inline features/step_definitions/salt_steps.rb]: Aaaahh, I see! Thank you for explaining it. 👍

**Comment by MalloZup** [comment]
@meaksh  we need a backport for head too.

- [ ] backport to head

**Comment by meaksh** [comment]
Merged on `manager31` and `master`.

**Comment by MalloZup** [comment]
ok thx @meaksh :rose:

**Comment by Bischoff** [comment]
Yes, please use the default timeout. It is very long (10 minutes), but as long as you can exit the loop earlier, that's fine, it will be so long only in case of an unrecoverable failure.

**Pull request merged**